### DISCUSSION TITLE
`ComboBox` named incorrectly? 😳 👉👈

### DISCUSSION THREAD

**Discussion by Enyium** [slint-ui/slint#6210] [General]
[Wikipedia](https://en.wikipedia.org/wiki/Combo_box) says on combo boxes:

> it is a combination of a [drop-down list](https://en.wikipedia.org/wiki/Drop-down_list) or [list box](https://en.wikipedia.org/wiki/List_box) and a single-line editable [textbox](https://en.wikipedia.org/wiki/Text_box), allowing the user to either type a value directly or select a value from the list.

Slint's [`ComboBox`](https://releases.slint.dev/latest/docs/slint/src/language/widgets/combobox.html) doesn't have a text box. So, there's no combinatorial aspect in it.

Wikipedia also says:

> The term "combo box" is sometimes used to mean "drop-down list".

But "sometimes" - colloquially, I'd think - doesn't seem to justify the name definition in a widget set of a UI framework.

I think the name `ComboBox` may only be justified, if you plan to extend the widget with the possibility of including a text box. But even then, maybe there should be two widgets, one of them named `ComboBox`, which would not be the one currently implemented.

**Comment by bjorn** [comment]
The combinatorial aspect of Slint's combo box is probably that it combines a button with a drop-down list, similarly to Qt's [QComboBox](https://doc.qt.io/qt-6/qcombobox.html). The latter does optionally allow the combo box to be editable, but I'm not sure if the lack of this feature is reason enough to argue the widget has the wrong name. I imagine Slint's ComboBox might eventually also support this, anyway.
